File: backend/src/main/java/com/qiankubx/module/expense/entity/Expense.java
package com.qiankubx.module.expense.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("t_expense")
public class Expense {

    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    private Long userId;

    private Long categoryId;

    private Long tripId;

    private Integer type;

    private BigDecimal amount;

    private BigDecimal taxAmount;

    private String invoiceNo;

    private String invoiceCode;

    private LocalDate invoiceDate;

    private String invoiceType;

    private String sellerName;

    private String buyerName;

    private String fileUrl;

    private String fileName;

    private String description;

    private LocalDate expenseDate;

    private Integer reimburseStatus;

    private Long reimbursementId;

    private String dataSign;

    private Integer status;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}

File: frontend/packages/shared/types/expense.ts
import type { PageQuery } from './common'

export interface InvoiceUploadVO {
  fileUrl: string
  fileName: string
  invoiceNo: string
  invoiceCode: string
  invoiceDate: string
  amount: number
  taxAmount: number
  sellerName: string
  buyerName: string
  invoiceType: string
  parsedSuccess: boolean
  parseSuccess: boolean
  parseMessage: string
}

export interface ExpenseCreateDTO {
  categoryId: number
  tripId?: number
  type: number
  amount: number
  taxAmount?: number
  invoiceNo?: string
  invoiceCode?: string
  invoiceDate?: string
  invoiceType?: string
  sellerName?: string
  buyerName?: string
  fileUrl?: string
  fileName?: string
  description?: string
  expenseDate: string
}

export interface ExpenseVO {
  id: number
  userId: number
  categoryId: number
  categoryName: string
  tripId: number
  type: number
  amount: number
  taxAmount: number
  invoiceNo: string
  invoiceCode: string
  invoiceDate: string
  invoiceType: string
  sellerName: string
  buyerName: string
  fileUrl: string
  fileName: string
  description: string
  expenseDate: string
  reimburseStatus: number
  reimbursementId: number
  dataSign: string
  status: number
  createdAt: string
  updatedAt: string
}

export interface ExpenseQueryDTO extends PageQuery {
  categoryId?: number
  type?: number
  reimburseStatus?: number
  startDate?: string
  endDate?: string
  page?: number
  pageSize?: number
}
